rts/System/float3.h

```diff
@@ -392,6 +392,18 @@ class float3
 		return (*this) * ca + axis.cross(*this) * sa + axis * axis.dot(*this) * (1.0f - ca);
 	}
 
+    /**
+     * @brief directional difference in rad
+     * @param other float3 to compute against
+     * @return float3 angle difference
+     *
+     * Calculates the angle between this float3
+     * and another float3
+     */
+    float3 angleDifference(const float3& other) const {
+         return float3::fmod((other - *this) + 3.0f*math::PI, 2.0f*math::PI) - math::PI;
```

Comment:
3.0f and 2.0f look a bit arbitrary.
Perhaps look into recipes here: https://stackoverflow.com/questions/2708476/rotation-interpolation